app/admin/dashboard.rb

```diff
@@ -47,6 +47,104 @@ def set_dashboard_values
           committed_one_month_ago: UserPractice.where(time_committed: @beginning_of_last_month..@end_of_last_month,committed: true).count,
           total_committed: UserPractice.where(committed: true).count
       }
+
+      @approved_enabled_published_practices = Practice.where(published: true, enabled: true, approved: true).order(Arel.sql("lower(practices.name) ASC"))
+
+      @practice_views_array = []
+      @site_visits_by_month = []
+      @total_users_by_month = []
+      @new_users_by_month = []
+      @month_and_year_array = []
+      start_date = Date.today.prev_year.beginning_of_month
+      end_date = Date.today
+
+      while start_date <= end_date
+        beg_of_month = start_date.beginning_of_day
+        end_of_month = start_date.end_of_month.end_of_day
+        month_and_year = Date::MONTHNAMES[start_date.month] + " #{ start_date.year.to_s}"
+        # Get practice views by month
+        @approved_enabled_published_practices.each do |p|
+          practice_visits_by_month = []
+          pr_visit_ct = Ahoy::Event.where_props(practice_id: p[:id]).where(time: beg_of_month...end_of_month).count
+          practice_visits_by_month << pr_visit_ct
+          practice_visits_by_month.each do |visit_count|
+            @practice_views_array << [p.id, visit_count]
+          end
+        end
+
+        # Get site visits by month
+        site_visit_by_ip = Ahoy::Event.where(name: 'Site visit').where("properties->>'ip_address' is not null").where(time: beg_of_month..end_of_month).group("properties->>'ip_address'").count
```

Comment:
I think we are reusing this somewhere else in the file. Can we make this into a method that we can call?

Reply:
I created a method for this. Thank you!